tensorflow/core/kernels/mkl_matmul_op_fused.cc

```diff
@@ -0,0 +1,194 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+// See docs in ../ops/math_ops.cc.
+
+// This file uses MKL-DNN InnerProduct for acceleration of TF Matrix-Matrix
+// Multiplication (MatMul) with bias (BiasAdd) operations.
+#ifdef INTEL_MKL
+
+#include "tensorflow/core/framework/register_types.h"
+#include "tensorflow/core/kernels/fill_functor.h"
+#include "tensorflow/core/kernels/mkl_matmul_ops_common.h"
+#include "tensorflow/core/lib/core/errors.h"
+
+namespace tensorflow {
+
+// Fuse Operation
+template <typename Device, typename T>
+class MklFusedMatMulOp : public MklDnnMatMulOpBase<T> {
+ public:
+  explicit MklFusedMatMulOp(OpKernelConstruction* ctx)
+      : MklDnnMatMulOpBase<T>(ctx) {
+    std::vector<string> fused_ops;
+
+    OP_REQUIRES_OK(ctx, ctx->GetAttr("fused_ops", &fused_ops));
+    OP_REQUIRES_OK(ctx, ctx->GetAttr("transpose_a", &transpose_a_));
+    OP_REQUIRES_OK(ctx, ctx->GetAttr("transpose_b", &transpose_b_));
+
+    OP_REQUIRES(ctx, fused_ops.size() == 1,
+                errors::InvalidArgument(
+                    "MklFusedMatMul must have only one argument: bias."));
+    OP_REQUIRES(
+        ctx, transpose_a_ == false,
+        errors::InvalidArgument("In[0] of MklMatMul can't be transposed."));
+  }
+
+  void Compute(OpKernelContext* ctx) override {
+    // FusedMatMul has 3 inputs: src, weights, bias
+    const Tensor& src_tensor = ctx->input(this->kInputIndexSrc);
+    const Tensor& weight_tensor = ctx->input(this->kInputIndexWeight);
+    const Tensor& bias_tensor = MklGetInput(ctx, this->kInputIndexBias);
+
+    MklDnnShape src_dnn_shape;
```

Comment:
Using `mkl` instead of `dnn` makes it clearer. Same for other places.
```suggestion
    MklDnnShape src_mkl_shape;
```